fendermint/vm/topdown/src/observation.rs

```diff
@@ -0,0 +1,123 @@
+// Copyright 2022-2024 Protocol Labs
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use crate::{BlockHeight, Bytes};
+use anyhow::anyhow;
+use arbitrary::Arbitrary;
+use fendermint_crypto::secp::RecoverableECDSASignature;
+use fendermint_crypto::SecretKey;
+use fendermint_vm_genesis::ValidatorKey;
+use serde::{Deserialize, Serialize};
+use std::fmt::{Display, Formatter};
+
+/// The content that validators gossip among each other.
```

Comment:
This isn't the gossiped message, but rather contained in the `CertifiedObservation` message, which was documented with "A self-certified observation made by a validator", which seems like the reverse.